alf/networks/network.py

```diff
@@ -126,6 +126,9 @@ def _test_forward(self):
         """
         inputs = common.zero_tensor_from_nested_spec(self._input_tensor_spec,
                                                      batch_size=2)
+        inputs = alf.nest.map_structure(
+            lambda x: x.to(torch.float32)
+            if x.dtype.is_floating_point else x, inputs)
```

Comment:
Shouldn't we expect `self.forward` can handle other dtypes?

Reply:
Some time the input is fp16. But to run it, the code need to be under autocast context. So here I just to change all the floating value to float32.

Reply:
Currently,  alf only use fp32 for the weights. There is no place to set dtype for weights. And the default dtype for alf is float32